### PR TITLE
basic settings modal

### PR DESCRIPTION
**Pull request by shriyan44**
extras:
- deleted fakedata file
- set signup password to show default

### CODE REVIEW

File: components/Profile/SettingsPopup.jsx [hunk 1]
```diff
@@ -1,54 +1,135 @@
...
+// };
 
-SettingsPopup.defaultProps = {
-  style: [],
-  triggerSize: 18,
-};
+// SettingsPopup.defaultProps = {
+//   style: [],
+//   triggerSize: 18,
+// };
```
Comment (jpoist97) [inline components/Profile/SettingsPopup.jsx]: I think you need to put back in triggerSize if you're gonna use it

File: components/Profile/SettingsPopup.jsx [hunk 2]
```diff
@@ -1,54 +1,135 @@
...
+import React, { useState } from 'react';
 import {
-  Menu, MenuOptions, MenuOption, MenuTrigger,
-} from 'react-native-popup-menu';
-import Proptypes from 'prop-types';
+  Alert, Modal, StyleSheet, Pressable, View, TouchableOpacity, Image,
+} from 'react-native';
 import styled from 'styled-components/native';
 import { Ionicons } from '@expo/vector-icons';
+import firebase from 'firebase';
```
Comment (jpoist97) [inline components/Profile/SettingsPopup.jsx]: Can we rename this file to SettingsModal.jsx

File: components/Profile/SettingsPopup.jsx [hunk 3]
```diff
@@ -1,54 +1,135 @@
...
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: 10,
+  },
+  modalView: {
+    height: 500,
+    width: 350,
+    margin: 10,
+    backgroundColor: 'white',
+    borderRadius: 20,
```
Comment (jpoist97) [inline components/Profile/SettingsPopup.jsx]: This should be done using styled components

File: components/Profile/SettingsPopup.jsx [hunk 4]
```diff
@@ -1,54 +1,135 @@
...
-          backgroundColor: '#EDEDF0', borderRadius: 10, padding: 5, width: 'auto',
-        },
-      }}
+    <View>
+      <Modal
+        animationType="slide"
+        transparent
+        visible={modalVisible}
+        onRequestClose={() => {
+          Alert.alert('Modal has been closed.');
```
Comment (jpoist97) [inline components/Profile/SettingsPopup.jsx]: What is this for?

File: components/Profile/SettingsPopup.jsx [hunk 5]
```diff
@@ -1,54 +1,135 @@
...
-              <StyledText>{option.text}</StyledText>
-            </Text>
-          </MenuOption>
-        ))}
-
-      </MenuOptions>
-    </Menu>
+
+        <TouchableOpacity style={styles.centeredView} onPress={() => setModalVisible(!modalVisible)}>
+          <TouchableOpacity style={styles.modalView} onPress={() => setModalVisible(modalVisible)}>
```
Comment (jpoist97) [inline components/Profile/SettingsPopup.jsx]: Why are these touchableOpacities? Do Views not work for these and why do you need the onPress stuff?

File: components/Profile/SettingsPopup.jsx [hunk 6]
```diff
@@ -1,54 +1,135 @@
...
+            <TouchableOpacity style={styles.colorButton} onPress={() => { alert('clicked blue'); }}>
+              <Image source={require('../../assets/blue.png')} />
+            </TouchableOpacity>
+            <TouchableOpacity style={styles.colorButton} onPress={() => { alert('clicked purple'); }}>
+              <Image source={require('../../assets/purple.png')} />
+            </TouchableOpacity>
+            <Pressable
+              style={[styles.button]}
+              onPress={async () => {
+                setModalVisible(false);
```
Comment (jpoist97) [inline components/Profile/SettingsPopup.jsx]: I think we should do this after they are successfully logged out

**Review by jpoist97 — DISMISSED** [review]
This is pretty much done, I left some questions I had but honestly I can just fix those things when I add to the modal cause I wanna start working on it soon. Another question I had was what's the difference between the Pressable and TouchableOpacity

File: components/Profile/SettingsPopup.jsx (hunk 5 above)
Comment (shriyan44) [reply components/Profile/SettingsPopup.jsx]: I had to use touchable opacity and on presses cuz its like the only workaround for making modal close when u touch out of it

**Review by finlaylp — APPROVED** [review]

**Comment by shriyan44** [comment]
> This is pretty much done, I left some questions I had but honestly I can just fix those things when I add to the modal cause I wanna start working on it soon. Another question I had was what's the difference between the Pressable and TouchableOpacity

theyre basically same but perusable has more optional props like a prop that defines how far a touch can register away from the the wrapped element so ill prolly use that for the small buttons

**Pull request merged**